provider/cloudflare/cloudflare_test.go

```diff
@@ -1561,132 +1555,18 @@ func TestCloudflareGroupByNameAndType(t *testing.T) {
-		{
-			Name:            "ProxiedByDefault: true, ProxiedRecord: true, ExpectUpdates: false",
-			SetupProvider:   func(p *CloudFlareProvider) { p.proxiedByDefault = true },
-			SetupRecord:     func(r *dns.RecordResponse) { r.Proxied = true },
-			ShouldBeUpdated: false,
-		},
-		{
-			Name:                "ProxiedByDefault: true, ProxiedRecord: false, ExpectUpdates: true",
-			SetupProvider:       func(p *CloudFlareProvider) { p.proxiedByDefault = true },
-			SetupRecord:         func(r *dns.RecordResponse) { r.Proxied = false },
-			ShouldBeUpdated:     true,
-			PropertyKey:         annotations.CloudflareProxiedKey,
-			ExpectPropertyValue: "true",
-		},
-		{
-			Name:                "ProxiedByDefault: false, ProxiedRecord: true, ExpectUpdates: true",
-			SetupProvider:       func(p *CloudFlareProvider) { p.proxiedByDefault = false },
-			SetupRecord:         func(r *dns.RecordResponse) { r.Proxied = true },
-			ShouldBeUpdated:     true,
-			PropertyKey:         annotations.CloudflareProxiedKey,
-			ExpectPropertyValue: "false",
-		},
-		// Comment tests
-		{
-			Name:            "DefaultComment: 'foo', RecordComment: 'foo', ExpectUpdates: false",
-			SetupProvider:   func(p *CloudFlareProvider) { p.DNSRecordsConfig.Comment = "foo" },
-			SetupRecord:     func(r *dns.RecordResponse) { r.Comment = "foo" },
-			ShouldBeUpdated: false,
-		},
-		{
-			Name:                  "DefaultComment: '', RecordComment: none, ExpectUpdates: true",
-			SetupProvider:         func(p *CloudFlareProvider) { p.DNSRecordsConfig.Comment = "" },
-			SetupRecord:           func(r *dns.RecordResponse) { r.Comment = "foo" },
-			ShouldBeUpdated:       true,
-			PropertyKey:           annotations.CloudflareRecordCommentKey,
-			ExpectPropertyPresent: false,
-		},
-		{
-			Name:                "DefaultComment: 'foo', RecordComment: 'foo', ExpectUpdates: true",
-			SetupProvider:       func(p *CloudFlareProvider) { p.DNSRecordsConfig.Comment = "foo" },
-			SetupRecord:         func(r *dns.RecordResponse) { r.Comment = "" },
-			ShouldBeUpdated:     true,
-			PropertyKey:         annotations.CloudflareRecordCommentKey,
-			ExpectPropertyValue: "foo",
-		},
-		// Regional Hostname tests
-		{
-			Name: "DefaultRegionKey: 'us', RecordRegionKey: 'us', ExpectUpdates: false",
-			SetupProvider: func(p *CloudFlareProvider) {
-				p.RegionalServicesConfig.Enabled = true
-				p.RegionalServicesConfig.RegionKey = "us"
-			},
-			RegionKey:       "us",
-			ShouldBeUpdated: false,
-		},
-		{
-			Name: "DefaultRegionKey: 'us', RecordRegionKey: 'us', ExpectUpdates: false",
-			SetupProvider: func(p *CloudFlareProvider) {
-				p.RegionalServicesConfig.Enabled = true
-				p.RegionalServicesConfig.RegionKey = "us"
-			},
-			RegionKey:           "eu",
-			ShouldBeUpdated:     true,
-			PropertyKey:         annotations.CloudflareRegionKey,
-			ExpectPropertyValue: "us",
-		},
-		// Custom Hostname tests
-		// TODO: add tests for custom hostnames when properly supported
```

Comment:
`TestProviderPropertiesIdempotency()` and `TestGroupByNameAndTypeWithCustomHostnames_MX()` got merged and all test cases removed.